### PR TITLE
Fix BC in NodeTestCase

### PR DESCRIPTION
**Pull request by remicollet**
For context see https://github.com/twigphp/Twig-extensions/issues/155

### CODE REVIEW

**Comment by remicollet** [comment]
Sorry, PR open against master... should probably be cherry-picked in 1.x

File: lib/Twig/Test/NodeTestCase.php [hunk 1]
```diff
@@ -15,17 +15,21 @@
     /**
      * @dataProvider getTests
      */
-    public function testCompile($node, $source, $environment = null)
+    public function testCompile($node, $source, $environment = null, $regex = false)
```
Comment (stof) [inline lib/Twig/Test/NodeTestCase.php]: the name is wrong. this is not a regex

**Comment by stof** [comment]
making the cache per user should be submitted as a separate PR. It seems unrelated to the BC break fix

File: lib/Twig/Test/NodeTestCase.php (hunk 1 above)
Comment (remicollet) [inline lib/Twig/Test/NodeTestCase.php]: what name do you want ?

Comment (stof) [inline lib/Twig/Test/NodeTestCase.php]: ``pattern`` may work. Or maybe ``isFormat`` (as PHPUnit calls it a format)

**Comment by remicollet** [comment]
Rewite and rebase for 1.x on PR #1905

**Pull request closed without merging**